pkg/azure/provider/helpers/driver.go

```diff
@@ -629,7 +634,33 @@ func createVMCreationParams(providerSpec api.AzureProviderSpec, imageRef armcomp
 		Zones:    getZonesFromProviderSpec(providerSpec),
 		Name:     &vmName,
 		Identity: getVMIdentity(providerSpec.Properties.IdentityID),
-	}, nil
+	}
+
+	// Processing for CVMs
+	if securityProfile := providerSpec.Properties.SecurityProfile; securityProfile != nil {
```

Comment:
Add a test in `provider_test.go` for successful creation of VM with security profile.

Reply:
Addressed in https://github.com/gardener/machine-controller-manager-provider-azure/pull/146/commits/cd74e3a151781e6cfec5f9cc2ef812cd8591d64c